syntax = "proto3";

package Storage;

service Storage {

  rpc getValues(User) returns (GetDataResponse) {}

  rpc setValues(UserValues) returns (Status) {}

  rpc removeUser(User) returns (Status) {}

  rpc addUser(User) returns (Status) {}

  rpc userIsExist(User) returns (IsExist) {}
}

message User {
  uint64 chat_id = 1;
  string name = 2;
}

message Value {
  string key = 1;
  float value = 2;
}

message UserValues {
  User user = 1;
  repeated Value values = 2;
}

message Status {
  bool is_success = 1;
  string message = 2;
}

message GetDataResponse {
  repeated Value values = 1;
  Status status = 2;
}

message IsExist {
  Status request_status = 1;
  bool is_exist = 2;
}
